iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeRealtimePriorityBlockingQueue.java

```diff
@@ -38,6 +39,12 @@ public class PipeRealtimePriorityBlockingQueue extends UnboundedBlockingPendingQ
   private final BlockingDeque<TsFileInsertionEvent> tsfileInsertEventDeque =
       new LinkedBlockingDeque<>();
 
+  private final AtomicInteger eventCount = new AtomicInteger(0);
+
+  private static final int pendingQueueConsumeThreshold = 1000;
```

Comment:
Better add these into commonConfig & pipeConfig and add configure logic from iotdb-system.properties. However, these shall not appear in iotdb-system.properties by default. Then it can be a hidden config for the convenience of online operation.